metecho/consumers.py

```diff
@@ -59,16 +59,29 @@ async def hydrate_message(self, content):
         # would cause every generic-message to include the user who's
         # getting the message. It'd just be noise on the wire.
         if model_name.lower() != "user":
-            try:
-                instance = await self.get_instance(model=model_name, id=id)
-            except ObjectDoesNotExist:
-                pass
+            # Handle single instances:
+            if id != "list":
+                try:
+                    instance = await self.get_instance(model=model_name, id=id)
+                except ObjectDoesNotExist:
+                    pass
+                else:
+                    content["payload"]["model"] = await database_sync_to_async(
+                        instance.get_serialized_representation
+                    )(self.scope["user"])
+            # Handle list events:
             else:
+                class_ = self.get_class(model=model_name)
                 content["payload"]["model"] = await database_sync_to_async(
-                    instance.get_serialized_representation
+                    class_.get_list_serialized_representation
```

Comment:
@wlonk I understand why this makes theoretical sense from the perspective of the list channel... but practically I'm trying to think through how to handle all this data that the front end doesn't actually need. All we _really_ need in the new `SCRATCH_ORG_RECREATE` event is the one org instance that has been re-created, even if the event is sent on the list channel. Getting the full list of orgs isn't necessary, and presents a problem for the front end to determine _which_ org was re-created. (The client never even queries or knows about the _full_ list of orgs -- it only fetches them per-task, on-demand.)

Even if we leave some of this logic in place, could we also allow sending custom events about one instance on the list channel?

Reply:
I was able to restructure this so you can subscribe to `list` and we can mark certain events as "for the list", but the individual object will still be serialized and sent. I think this should work.

I agree, this was theoretically pure but practically bad.